### ISSUE TITLE
Missing detailed IP block descriptions integrated in the JH7100 SoC

### ISSUE THREAD

**Issue by lbmeng** [starfive-tech/beaglev_doc#3]
It looks current version of the JH7100 datasheet is missing detailed IP block descriptions like registers.

Without that, it's hard for any upstream work like OpenSBI/U-Boot/Linux drivers.

Will future version datasheet include these?

**Comment by pdp7** [comment]
Yes, you are correct that the current datasheet is not sufficient to implement most drivers.   

Many of the 3rd party IP peripherals are able to use existing compatible strings in the [device tree](https://github.com/starfive-tech/sft-riscv-uboot/blob/starfive/arch/riscv/dts/starfive_vic7100_evb.dts).  The [u-boot repo](https://github.com/starfive-tech/sft-riscv-uboot/tree/starfive) and [linux repo](https://github.com/starfive-tech/sft-riscv-linux-5.10/tree/starfive) show what changes have been made to those upstream drivers.  It is mostly related to having to flush L2 cache as result of the [7100 bus architecture](https://github.com/starfive-tech/sft-riscv-linux-5.10/issues/2) (which is being solved in the 7110 SoC).  I agree that developers working on upstreaming need actual documentation and not just the existing downstream code.

This version of the 7100 datasheet is what could be done in time for the beta starting.  StarFive is continuing to work on the datasheet and should be releasing new versions as new sections become available.

FYI - there a discussion thread about documentation on the [BeagleV forum group](https://forum.beagleboard.org/t/datasheet-on-the-soc/28882/15).

**Comment by lbmeng** [comment]
Thanks Drew for the very useful information!

I will have to look into the StarFive fork of U-Boot/Linux repos to understand the IP blocks for now :)
